src/kv/encryptor.h

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "crypto/symmetric_key.h"
+#include "ds/spin_lock.h"
+#include "kv/kv_types.h"
+
+namespace kv
+{
+  class TxEncryptor : public kv::AbstractTxEncryptor
+  {
+  public:
+    struct KeyInfo
+    {
+      kv::Version version;
+
+      // This is unfortunate. Because the encryptor updates the ledger secrets
```

Comment:
@jumaffre Why is this unfortunate? Is it because we keep another copy of the ledger secrets somewhere else? If so, where?

Reply:
@achamayou AIUI, we also have a _useable_ representation of the key in-memory:

```
    struct EncryptionKey : KeyInfo
    {
      crypto::KeyAesGcm key;
    };
```

So its unfortunate that we need to store both a transferable raw representation, and the struct that operates with it.